openhands/sdk/conversation/impl/remote_conversation.py

```diff
@@ -523,6 +524,27 @@ def update_secrets(self, secrets: Mapping[str, SecretValue]) -> None:
         resp = self._client.post(f"/api/conversations/{self._id}/secrets", json=payload)
         resp.raise_for_status()
 
+    def generate_title(self, llm: LLM | None = None, max_length: int = 50) -> str:
+        """Generate a title for the conversation based on the first user message.
+
+        Args:
+            llm: Optional LLM to use for title generation. If not provided,
+                 uses self.agent.llm.
+            max_length: Maximum length of the generated title.
+
+        Returns:
+            A generated title for the conversation.
+
+        Raises:
+            ValueError: If no user messages are found in the conversation.
+        """
+        # Use provided LLM or fall back to agent's LLM
+        llm_to_use = llm or self.agent.llm
+
+        return generate_conversation_title(
+            events=self._state.events, llm=llm_to_use, max_length=max_length
+        )
+
```

Comment:
Can we add a related route in agent_server and remote conversation will make request to that endpoint?